cfme/infrastructure/virtual_machines.py

```diff
@@ -960,22 +1066,21 @@ def get_number_of_vms(do_not_navigate=False):
 @navigator.register(Template, 'All')
 @navigator.register(Vm, 'All')
 class VmAllWithTemplates(CFMENavigateStep):
-    prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
-
-    def step(self, *args, **kwargs):
-        self.prerequisite_view.navigation.select('Compute', 'Infrastructure', '/vm_infra/explorer')
-        accordion.tree('VMs & Templates', 'All VMs & Templates')
+    VIEW = VmsTemplatesAllView
+    prerequisite = NavigateToObject(Server, 'LoggedIn')
 
-    def resetter(self, *args, **kwargs):
-        reset_page()
+    def step(self):
+        self.prerequisite_view.navigation.select('Compute', 'Infrastructure', 'Virtual Machines')
+        self.view.sidebar.vmstemplates.tree.click_path('All VMs & Templates')
 
-    def am_i_here(self, *args, **kwargs):
-        return match_page(summary='All VMs & Templates')
+    def am_i_here(self):
+        return self.view.is_displayed and self.view.sidebar.vmstemplates.is_opened
 
 
 @navigator.register(Template, 'AllForProvider')
 @navigator.register(Vm, 'AllForProvider')
 class VmAllWithTemplatesForProvider(CFMENavigateStep):
+    VIEW = VmsTemplatesAllView
```

Comment:
@izapolsk @mfalesni @psav Tagging some other widgetastic dev's since this is an important consideration for many of our modules.

I don't think the 'All' view for any page model should have methods to navigate through a given widget to get to some detailed selection, as this will confuse page models with navigation destinations.

If these pages have different UI elements, then we need separate view classes to accurately represent them. The assertion for whether a page is currently displayed needs to be collapsed down into the view's `is_displayed` method otherwise it will be very unreliable to use outside of trying to navigate again.  We need to be able to create a view object and reliably assert `view.is_displayed` - we can't try to call some navigation class's `am_i_here` directly to assert what page we're at.

I don't think we necessarily need to provide navigation destinations for Archived/Orphaned and other folders within the accordion menus, especially if their views are not unique and share toolbar/entities with the view that is used for the other folders within the specific accordion.  This may create a situation where I navigate to 'Vms & Templates' accordion, and then use the returned view to call `view.tree.click_path('Archived')` to get to the archived folder.  Now - if I call view.is_displayed() at this point, should it return False since I'm not in the 'All Vms & Templates' folder, or should it return True since its still in 'Vms & Templates'?  I believe it should return False - though the view can still be used to interact with the toolbar/entities after selecting a different option in the tree.

I do think that we need to model separate views for these pages when there are differences in their toolbars/entities, and add a navigation destination that returns this view if a test needs it.
In cloud.instances, there are navigation destinations for the 4 accordion blocks, but not to the Archived/Orphaned sections within the block.

We haven't provided navigation destinations to the filter folders within the 'All' views anywhere, for example.